SingularityUI/app/components/common/modalButtons/ShuffleOptOutModal.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import FormModal from '../modal/FormModal';
+import { EnableRequestShuffleOptOut, DisableRequestShuffleOptOut } from '../../../actions/api/requests.es6';
+
+class ShuffleOptOutModal extends Component {
+
+  static propTypes = {
+    requestId: PropTypes.oneOfType([PropTypes.string, PropTypes.array]).isRequired,
+    isOptedOut: PropTypes.bool.isRequired,
+    saveShuffleOptOut: PropTypes.func.isRequired,
+    then: PropTypes.func
+  };
+
+  show() {
+    this.refs.shuffleOptOutModal.show();
+  }
+
+  confirm(data) {
+    const requestIds = typeof this.props.requestId === 'string' ? [this.props.requestId] : this.props.requestId;
+    for (const requestId of requestIds) {
+      this.props.saveShuffleOptOut(requestId, data, [409]);
+    }
+  }
+
+  renderActionText() {
+    if (this.props.isOptedOut) {
+      return <strong>Enable</strong>
+    } else {
+      return <strong>Disable</strong>
+    }
+  }
+
+  render() {
+    const requestIds = typeof this.props.requestId === 'string' ? [this.props.requestId] : this.props.requestId;
+    const { isOptedOut } = this.props;
+
+    return (
+      <FormModal
+        name={isOptedOut ? "Enable Shuffling For Request" : "Disable Shuffling For Request"}
+        ref="shuffleOptOutModal"
+        action={isOptedOut ? "Enable Shuffling" : "Disable Shuffling"}
+        onConfirm={(data) => this.confirm(data)}
+        buttonStyle="primary"
+        formElements={[]}>
+        <p>{this.renderActionText()} shuffling for {requestIds.length > 1 ? 'these' : 'this'} request{requestIds.length > 1 && 's'}?</p>
+        <p>Note that Singularity may still bounce your tasks if necessary, such as during host failures.</p>
```

Comment:
Two things, we can probably add a first note that this is specifically 'Disable task shuffling due to memory pressure or cpu overusage', with a follow up of 'Note that this does not prevent all restarts of your task. A task may still be bounced or shuffled for events such as host decommissions'

Reply:
![image](https://user-images.githubusercontent.com/16820795/73090674-ce9b5100-3ea6-11ea-8468-2b932b3ca388.png)

Reply:
Looks good, let's give this a go in QA 👍